modules/common/src/main/scala/notebook/front/widgets/package.scala

```diff
@@ -506,6 +506,16 @@ package object widgets {
                                                   "width" → sizes._1, "height" → sizes._2)))
   }
 
+  case class CustomC3Chart[C:ToPoints:Sampler](originalData:C, chartOptions :String = "{}", override val sizes:(Int, Int)=(600, 400), maxPoints:Int = 25) extends Chart[C] {
```

Comment:
not sure. 

- `JsValue` is Play-specific, so people would get scared ;) anyway they'd most likely [create it by parsing  regular json](https://www.playframework.com/documentation/2.2.x/ScalaJson#Using-string-parsing).
- maybe something similar could be useful for defining & combining a few preset `configurations`, like `stacked bar chart` with custom options like `column-name for x-axis`. But not convinced yet.